### PR TITLE
Change core-js exports to use named exports. And make exports ts friendly

### PR DESCRIPTION
**Pull request by imeredith**
# Description

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by imeredith** [comment]
@sophistifunk will be better when its finished. Spent many hours dealing with js-builder bs to make this work

File: blueocean-core-js/src/js/fetch.ts [hunk 1]
```diff
@@ -2,56 +2,60 @@ import * as Promise from 'bluebird';
...
     return !fetchOptions || !fetchOptions.method || 'get'.localeCompare(fetchOptions.method) === 0;
 }
-interface RawFetchOpts {
-    onSuccess?: <A, B>(success: A) => B
-    onError?: <A, B>(error: A) => B
-    fetchOptions?: RequestInit
-    disableDedupe?: boolean
-    disableLoadingIndicator?: boolean
-    ignoreRefreshHeader?: boolean
+export namespace FetchFunctions {
```
Comment (sophistifunk) [inline blueocean-core-js/src/js/fetch.ts]: Namespaces FTW

**Comment by sophistifunk** [comment]
I wonder if anybody's done a study into the correlation between commit message length and developer "sick of this shit" level?

**Comment by sophistifunk** [comment]
We need some sort of auto-xkcd github bot.

https://imgs.xkcd.com/comics/git_commit_2x.png

**Comment by sophistifunk** [comment]
Sweeeeeeeet

**Pull request merged**